session/bootstrap.go

```diff
@@ -1288,6 +1296,76 @@ func upgradeToVer59(s Session, ver int64) {
 	writeOOMAction(s)
 }
 
+func upgradeToVer60(s Session, ver int64) {
+	if ver >= version60 {
+		return
+	}
+	h := &bindinfo.BindHandle{}
+	var err error
+	mustExecute(s, "BEGIN PESSIMISTIC")
+
+	defer func() {
+		if err != nil {
+			mustExecute(s, "ROLLBACK")
+			return
+		}
+
+		mustExecute(s, "COMMIT")
+	}()
+	mustExecute(s, h.LockBindInfoSQL())
```

Comment:
We don't need this locking in bootstrap, and `lockBindInfoSQL` does not need to be exported then?

Reply:
Will multiple TiDB instances upgrade at the same time have concurrency issues?